fail_safe_firmware/FS_MCU_Arduino_Nano/FS_MCU_Arduino_Nano.ino

```diff
@@ -0,0 +1,159 @@
+  
+  pinMode(outD6, OUTPUT);
+  pinMode(outD7, OUTPUT);
+  pinMode(outD8, OUTPUT);
+  pinMode(outD9, OUTPUT);
+  pinMode(outD10, OUTPUT);
+  pinMode(outD11, OUTPUT);
+  pinMode(outD12, OUTPUT);
+
+}
+
+void loop() {
+  while (true){
+    armed = digitalRead(inpA0);
+    if (armed){
+      digitalWrite(outD6,HIGH);   //LED Armed HIGH
+    }
+    else{
+      digitalWrite(outD6,LOW);    //LED Armed LOW
+    }
+        
+    if (FS_locked){
+      
+      //try to unlock KS
+      if (armed){
+        continue;
+      }
+      else{
+        FS_locked = false;
+        digitalWrite(outD3,LOW);   //LED Locked LOW
+        continue;
+      }
+    }
+    
+    //KS not locked
+    else{
+      //OtA KS test
+      if (digitalRead(inpA1)){
+        digitalWrite(outD11,HIGH);    //LED OtA KS HIGH
+        
+        //HW KS test
+        if(digitalRead(inpA2)){
+          digitalWrite(outD11,HIGH);    //LED HW KS HIGH
+          
+          //SW KS test
+          if (digitalRead(inpA3)){
+            digitalWrite(outD9,HIGH);    //LED SW KS HIGH
+            digitalWrite(outD2,LOW);    //MCU mosfet --- MAIN OUTPUT set LOW
+            all_sys_go = false;
+            continue;
+          }
+          else{
+            digitalWrite(outD9,LOW);    //LED SW KS LOW
+            if(armed){
+              //Set MAIN OUTPUT HIGH
+              digitalWrite(outD2,HIGH);    //MCU mosfet --- MAIN OUTPUT set HIGH
+              all_sys_go = true;
+              continue;
+            }
+            else{
+              digitalWrite(outD2,LOW);    //MCU mosfet --- MAIN OUTPUT set LOW
+              all_sys_go = false;
+              continue;
+            }
+          }
+        }
+        //HW KS low
+        else{
+          digitalWrite(outD2,LOW);    //MCU mosfet --- MAIN OUTPUT set LOW
+          digitalWrite(outD11,LOW);    //LED HW KS LOW
+          FS_locked = true;
+          all_sys_go = false;
+        }
+      }
+      //OtA KS low
+      else{
+        digitalWrite(outD11,LOW);    //LED OtA KS LOW
+        digitalWrite(outD2,LOW);    //MCU mosfet --- MAIN OUTPUT set LOW
+        FS_locked = true;
+      }    
+    }
+    status_lights();
+  }
+}
+
+void status_lights(){
+  if (all_sys_go){
+    if(digitalRead(inpA5)){
+      //Manual
+      digitalWrite(outD12,LOW);    //RED status pin 
+      digitalWrite(outD8,LOW);    //GREEN status pin 
+      digitalWrite(outD7,LOW);    //BLUE status pin 
+      digitalWrite(outD4,HIGH);    //YELLOW status pin 
+    }
+    else{
+      //Autonomous
+      digitalWrite(outD12,LOW);    //RED status pin 
+      digitalWrite(outD8,HIGH);    //GREEN status pin 
+      digitalWrite(outD7,LOW);    //BLUE status pin 
+      digitalWrite(outD4,LOW);    //YELLOW status pin 
+    }
+  }
+  else{
+    //KS triggered
+    digitalWrite(outD12,HIGH);    //RED status pin 
+    digitalWrite(outD8,LOW);    //GREEN status pin 
+    digitalWrite(outD7,LOW);    //BLUE status pin 
+    digitalWrite(outD4,LOW);    //YELLOW status pin 
```

Comment:
I believe this can be simplified using a setRGBY() helper. Something along the lines of:
```
void setRGBY(bool R, bool G, bool B, bool Y) {
      digitalWrite(outD12, R); 
      digitalWrite(outD8, G);   
      digitalWrite(outD7, B); 
      digitalWrite(outD4, Y); 
}
```
Alternatively modify this to work with variable inputs?

Reply:
Something along the lines of https://stackoverflow.com/questions/1472138/c-default-arguments